From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Fri, 3 Jun 2022 17:17:50 -0700
Subject: [format] Add a few prettier-ignores to work around a Prettier
 instability bug

Discussion with a reduced repro of the Prettier bug:
  https://github.com/zulip/zulip-mobile/pull/5396#issuecomment-1146470871
---
 types/@react-navigation/core/lib/typescript/src/types.js.flow   | 1 +
 .../routers/lib/typescript/src/StackRouter.js.flow              | 2 ++
 .../routers/lib/typescript/src/TabRouter.js.flow                | 1 +
 3 files changed, 4 insertions(+)

diff --git types/@react-navigation/core/lib/typescript/src/types.js.flow types/@react-navigation/core/lib/typescript/src/types.js.flow
index 600567b7a..f870405db 100644
--- types/@react-navigation/core/lib/typescript/src/types.js.flow
+++ types/@react-navigation/core/lib/typescript/src/types.js.flow
@@ -167,6 +167,7 @@ type NavigationHelpersCommon<
   State: NavigationState<> = NavigationState<>,
 > = {
   dispatch(action: NavigationAction | ((state: State) => NavigationAction)): void,
+  // prettier-ignore
   navigate<RouteName: $Keys<ParamList>>(
     ...args: $FlowFixMe /*  undefined extends ParamList[RouteName] ? [screen: RouteName] | [screen: RouteName, params: ParamList[RouteName]] : [screen: RouteName, params: ParamList[RouteName]] */ /* tsflower-unimplemented: ConditionalType */
   ): void,
diff --git types/@react-navigation/routers/lib/typescript/src/StackRouter.js.flow types/@react-navigation/routers/lib/typescript/src/StackRouter.js.flow
index 756ae8ed4..ad878904e 100644
--- types/@react-navigation/routers/lib/typescript/src/StackRouter.js.flow
+++ types/@react-navigation/routers/lib/typescript/src/StackRouter.js.flow
@@ -55,9 +55,11 @@ export type StackNavigationState<ParamList: ParamListBase> = NavigationState<Par
 };
 
 export type StackActionHelpers<ParamList: ParamListBase> = {
+  // prettier-ignore
   replace<RouteName: $Keys<ParamList>>(
     ...args: $FlowFixMe /*  undefined extends ParamList[RouteName] ? [screen: RouteName] | [screen: RouteName, params: ParamList[RouteName]] : [screen: RouteName, params: ParamList[RouteName]] */ /* tsflower-unimplemented: ConditionalType */
   ): void,
+  // prettier-ignore
   push<RouteName: $Keys<ParamList>>(
     ...args: $FlowFixMe /*  undefined extends ParamList[RouteName] ? [screen: RouteName] | [screen: RouteName, params: ParamList[RouteName]] : [screen: RouteName, params: ParamList[RouteName]] */ /* tsflower-unimplemented: ConditionalType */
   ): void,
diff --git types/@react-navigation/routers/lib/typescript/src/TabRouter.js.flow types/@react-navigation/routers/lib/typescript/src/TabRouter.js.flow
index f74b447c5..c159ffde6 100644
--- types/@react-navigation/routers/lib/typescript/src/TabRouter.js.flow
+++ types/@react-navigation/routers/lib/typescript/src/TabRouter.js.flow
@@ -40,6 +40,7 @@ export type TabNavigationState<ParamList: ParamListBase> = $Diff<
 };
 
 export type TabActionHelpers<ParamList: ParamListBase> = {
+  // prettier-ignore
   jumpTo<RouteName: Extract<$Keys<ParamList>, string>>(
     ...args: $FlowFixMe /*  undefined extends ParamList[RouteName] ? [screen: RouteName] | [screen: RouteName, params: ParamList[RouteName]] : [screen: RouteName, params: ParamList[RouteName]] */ /* tsflower-unimplemented: ConditionalType */
   ): void,
-- 
2.32.0

